File: 1/day1.py
"""
Day 1:
    Puzzle input    -> List of integers representing the number of calories each food contains
                    -> New line indicates that a new elf is being looked at

    Task 1  -> How many calories is the elf that is holding the most calories holding?
    Task 2  -> What is the total calorie count for the top 3 elves?
"""
#TASK 1============================================================================
#Open file containing puzzle inputs
with open("input.txt","r") as f:

    elf_list = []   #List of total calories being held by each elf
    calories = 0    #Calorie counter of each elf
    for line in f:
        #Checks if new line
        if line == "\n":
            elf_list.append(calories)
            calories = 0
        else:
            calories += int(line)

print(f"The elf carrying the most calories is carrying {max(elf_list)} calories")

#Sorts the list of elves in om ascending order
elf_list.sort()

#Gets the 1st, 2nd, and 3rd highest elf
first = elf_list[-1]
second = elf_list[-2]
third = elf_list[-3]

#Prints results
print()
print("Top 3 elves:")
print(f"1st: {first}")
print(f"2nd: {second}")
print(f"3rd: {third}")
print()
print(f"Total of top 3: {first+second+third}")


        


        

File: 1/day1.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <vector>
#include <algorithm>

int main() {
	//Opens input file
	std::fstream input("input.txt");

	//Checks if file is open
	if (!input.is_open())
		return EXIT_FAILURE;

	//Initialise variables
	std::vector<int> totals;	//Contain total calories for each elf
	int calories = 0; //Calorie counter
	std::string line;	//Line from input file
	while (std::getline(input,line)) {
		if (line == "") {
			totals.push_back(calories);
			calories = 0;
			continue;
		}
		//Converts string into integer
		calories += std::stoi(line);
	}

	//Sorts vector into ascending order
	std::sort(totals.begin(), totals.end());

	std::cout << "The elf with the most food has: ";
	std::cout << totals.back() << std::endl;
	
	//Counts the last values then removes it to get the top 3
	int counter = 0;
	for (int i = 0;i < 3;i++) {
		counter += totals.back();
		totals.pop_back();
	}
	std::cout << "The top 3 elves have a total calorie count of: " << counter << std::endl;

	return 0;
}